decoder/vaapidecoder_mpeg2.cpp

```diff
@@ -543,34 +544,61 @@ YamiStatus VaapiDecoderMPEG2::fillConfigBuffer()
+                            | (m_sequenceHeader->vertical_size_value & 0xFFF);
 
-    m_configBuffer.profile = m_VAProfile;
+    if (m_VAStart) {
+        // sequence start extension code with VA started has to conduct a
+        // port reconfiguration when stream size changes
+        if (m_configBuffer.width > m_videoFormatInfo.width
+            || m_configBuffer.height > m_videoFormatInfo.height) {
+            // need to re-start VA to generate new surfaces
+            status = VaapiDecoderBase::terminateVA();
+            if (status != YAMI_SUCCESS) {
+                return status;
+            }
+            m_VAStart = false;
+        }
+        else if (m_configBuffer.width != m_videoFormatInfo.width
+                 || m_configBuffer.height != m_videoFormatInfo.height) {
+            // VA surfaces can be re-used and client is notified
+            // public member from base class
+            m_videoFormatInfo.width = m_configBuffer.width;
+            m_videoFormatInfo.height = m_configBuffer.height;
+            m_videoFormatInfo.surfaceWidth = m_configBuffer.width;
+            m_videoFormatInfo.surfaceHeight = m_configBuffer.height;
+            m_configBuffer.surfaceWidth = m_configBuffer.width;
+            m_configBuffer.surfaceHeight = m_configBuffer.height;
+            status = YAMI_DECODE_FORMAT_CHANGE;
+        }
+    }
 
-    DEBUG("MPEG2: start() buffer size: %d x %d m_VAProfile %x level %x",
-          m_configBuffer.width, m_configBuffer.height, m_configBuffer.profile,
-          level);
+    if (!m_VAStart) {
 
-    m_configBuffer.surfaceWidth = m_configBuffer.width;
-    m_configBuffer.surfaceHeight = m_configBuffer.height;
-    if (m_configBuffer.surfaceNumber) {
-        m_configBuffer.surfaceNumber = m_configBuffer.surfaceNumber;
-    } else {
-        m_configBuffer.surfaceNumber = 8;
-    }
-    // no information is sent to start libva at this point
-    m_configBuffer.data = NULL;
-    m_configBuffer.size = 0;
+        m_configBuffer.profile = m_VAProfile;
 
-    // ready to start libva
-    status = VaapiDecoderBase::start(&m_configBuffer);
-    if (status != YAMI_SUCCESS) {
-        return status;
+        DEBUG("MPEG2: start() buffer size: %d x %d m_VAProfile %x level %x",
+              m_configBuffer.width, m_configBuffer.height,
+              m_configBuffer.profile, level);
+
+        m_configBuffer.surfaceWidth = m_configBuffer.width;
+        m_configBuffer.surfaceHeight = m_configBuffer.height;
+        if (!m_configBuffer.surfaceNumber) {
+            m_configBuffer.surfaceNumber = kMinSurfaces;
+        }
+        // no information is sent to start libva at this point
+        m_configBuffer.data = NULL;
+        m_configBuffer.size = 0;
+
+        // ready to start libva
+        status = VaapiDecoderBase::start(&m_configBuffer);
+        if (status != YAMI_SUCCESS) {
+            return status;
+        }
+        m_VAStart = true;
+        status = YAMI_DECODE_FORMAT_CHANGE;
```

Comment:
Just reminder,in long term we want return format change before we allocate surfaces. So the caller get a chance to do some preparation.Like ask surface from renderer.It's ok for current logical.